include/zephyr/drivers/i2c.h

```diff
@@ -785,6 +785,8 @@ static inline int z_impl_i2c_transfer(const struct device *dev,
 	const struct i2c_driver_api *api =
 		(const struct i2c_driver_api *)dev->api;
 
+	msgs[num_msgs - 1].flags |= I2C_MSG_STOP;
```

Comment:
Please add an assert before this verifying num_msgs >= 1

Reply:
ok, did not notice we allow a length of zero :) Out of curiosity, why do we allow this?